### PR TITLE
Add `ServiceWorkerGlobalScope.serviceWorker`

### LINKED ISSUE #29123: Missing property `ServiceWorkerGlobalScope.serviceWorker`
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

The ServiceWorkerGlobalScope.serviceWorker property is missing in mdn

### What did you expect to see?

Add a page for ServiceWorkerGlobalScope.serviceWorker and update ServiceWorkerGlobalScope summary page

### Do you have any supporting links, references, or citations?

https://w3c.github.io/ServiceWorker/#service-worker-global-scope-serviceworker

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/serviceworkerglobalscope`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerGlobalScope
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/serviceworkerglobalscope/index.md
* Last commit: https://github.com/mdn/content/commit/56c76424a5edb45f6716ac4ee48861dac8e7ae38
* Document last modified: 2023-08-04T07:12:32.000Z

</details>

### LINKED ISSUE #29541: The description of `ServiceWorker`'s getter is not comprehensive enough
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

```md
A ServiceWorker object is available in the `ServiceWorkerRegistration.active` property, and the `ServiceWorkerContainer.controller` property — this is a service worker that has been activated and is controlling the page (the service worker has been successfully registered, and the controlled page has been reloaded.)
```

The getter description of ServiceWorker only mentioned `ServiceWorkerRegistration.active` and `ServiceWorkerContainer.controller`

However, ServiceWorker may get via `ServiceWorkerGlobalScope.serviceWorker` and `ServiceWorkerRegistration.installing` also `ServiceWorkerRegistration.waiting`

### What did you expect to see?

Update this description to fully reflect ServiceWorker's getter

### Do you have any supporting links, references, or citations?

**Added yet**

https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/active

https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerContainer/controller

**Need Add**

See #29123 and #29461 

https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/installing

https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/waiting

**Related documentation**

https://w3c.github.io/ServiceWorker/#serviceworker-interface

https://w3c.github.io/ServiceWorker/#serviceworkerregistration-interface

https://w3c.github.io/ServiceWorker/#serviceworkercontainer-interface

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/serviceworker`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/serviceworker/index.md
* Last commit: https://github.com/mdn/content/commit/33fe6977c5dcbf4d5d89ed1faa5559a778936b65
* Document last modified: 2023-05-15T05:31:24.000Z

</details>

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

add `ServiceWorkerGlobalScope.serviceWorker` documentation

### Additional details

https://w3c.github.io/ServiceWorker/#serviceworkerglobalscope-interface

https://w3c.github.io/ServiceWorker/#service-worker-global-scope-serviceworker

### Related issues and pull requests

Fixes #29123 
Fixes https://github.com/mdn/content/issues/29541

### CODE REVIEW

**Comment by hamishwillee** [comment]
Flyby comment,
- YOu might add a trivial example, like this https://developer.mozilla.org/en-US/docs/Web/API/Navigator/serviceWorker#examples
- it seems odd that this accessor is not mentioned in https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker

**Comment by skyclouds2001** [comment]
> Flyby comment,
> 
> * YOu might add a trivial example, like this https://developer.mozilla.org/en-US/docs/Web/API/Navigator/serviceWorker#examples
> * it seems odd that this accessor is not mentioned in https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorker

For I, as ServiceWorkerGlobalScope: registration and ServiceWorkerGlobalScope: clients also do not provide an example, I think maybe it is not necessary to provide an example
For II, I'll create another pull request to add the additonal content

**Comment by hamishwillee** [comment]
> For I, I think as it is a simple getter, maybe it is not quite necessary to provide an example

I've had reviewers ask for this of me because readers like to see an example - it makes copy-paste easy and confirms that the code to use the API will indeed be what they think it is.

As a fly-by reviewer I'm explicitly removing myself as the person that places requirements on you - I'm just adding comments. BUT, if you do add an example it should be done in this PR and so should the ServiceWorker docs - either it is related an important and it should be done as the same work, or it isn't and can be ignored for the PR. But doing it in another PR just increases the chance it doesn't get done.

**Comment by skyclouds2001** [comment]
As [#29451](https://github.com/mdn/content/issues/29541) seems to not be working, that issue will fix in this pull request

**Comment by skyclouds2001** [comment]
> > For I, I think as it is a simple getter, maybe it is not quite necessary to provide an example
> 
> I've had reviewers ask for this of me because readers like to see an example - it makes copy-paste easy and confirms that the code to use the API will indeed be what they think it is.
> 
> As a fly-by reviewer I'm explicitly removing myself as the person that places requirements on you - I'm just adding comments. BUT, if you do add an example it should be done in this PR and so should the ServiceWorker docs - either it is related an important and it should be done as the same work, or it isn't and can be ignored for the PR. But creating another PR just increases the chance it doesn't get done.

Thanks for your suggestion!

**Comment by hamishwillee** [comment]
@wbamberg If you don't look this week I will try to next week.

File: files/en-us/web/api/serviceworkerglobalscope/serviceworker/index.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Value
+
+A {{domxref("ServiceWorker")}} object.
+
+## Examples
+
+```js
+self.serviceWorker;
+```
```
Comment (skyclouds2001) [inline files/en-us/web/api/serviceworkerglobalscope/serviceworker/index.md:21]:
```suggestion
## Syntax

```js-nolint
self.serviceWorker
```
```

File: files/en-us/web/api/serviceworkerglobalscope/registration/index.md [hunk 2]
```diff
@@ -14,6 +14,12 @@ The **`registration`** read-only property of the {{domxref("ServiceWorkerGlobalS
 
 A {{domxref("ServiceWorkerRegistration")}} object.
 
+## Examples
+
+```js
+self.registration;
+```
```
Comment (skyclouds2001) [inline files/en-us/web/api/serviceworkerglobalscope/registration/index.md:21]:
```suggestion
## Syntax

```js-nolint
self.registration
```
```

File: files/en-us/web/api/serviceworkerglobalscope/clients/index.md [hunk 3]
```diff
@@ -16,6 +16,12 @@ object associated with the service worker.
 
 The {{domxref("Clients")}} object associated with the specific worker.
 
+## Examples
+
+```js
+self.clients;
+```
```
Comment (skyclouds2001) [inline files/en-us/web/api/serviceworkerglobalscope/clients/index.md:23]:
```suggestion
## Syntax

```js-nolint
self.clients
```
```

File: files/en-us/web/api/serviceworkerglobalscope/clients/index.md [hunk 4]
```diff
@@ -16,6 +16,11 @@ object associated with the service worker.
 
 The {{domxref("Clients")}} object associated with the specific worker.
 
+## Syntax
+
+```js-nolint
+self.clients
+
```
Comment (skyclouds2001) [inline files/en-us/web/api/serviceworkerglobalscope/clients/index.md:23]:
```suggestion
## Syntax

```js-nolint
self.clients
```

```

**Comment by skyclouds2001** [comment]
cc @hamishwillee would you please review this if available

File: files/en-us/web/api/serviceworker/index.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ browser-compat: api.ServiceWorker
...
 
-A `ServiceWorker` object is available in the {{domxref("ServiceWorkerRegistration.active")}} property, and the {{domxref("ServiceWorkerContainer.controller")}} property — this is a service worker that has been activated and is controlling the page (the service worker has been successfully registered, and the controlled page has been reloaded.)
+A `ServiceWorker` object is available via the {{domxref("ServiceWorkerRegistration.active")}} property, the {{domxref("ServiceWorkerGlobalScope.serviceWorker")}} property and the {{domxref("ServiceWorkerContainer.controller")}} property when the service worker is in `activating` or `activated` state, and via {{domxref("ServiceWorkerRegistration.installing")}} property when the service worker is in `installing` state, or via {{domxref("ServiceWorkerRegistration.waiting")}} property when the service worker is in `installed` state.
```
Comment (hamishwillee) [inline files/en-us/web/api/serviceworker/index.md:12]: Let's chop this up to be more readable.

```suggestion
A `ServiceWorker` object is available via a number of propertes:

 - {{domxref("ServiceWorkerRegistration.active")}}
 - {{domxref("ServiceWorkerGlobalScope.serviceWorker")}}
 - {{domxref("ServiceWorkerContainer.controller")}} — when the service worker is in `activating` or `activated` state
 - {{domxref("ServiceWorkerRegistration.installing")}} — when the service worker is in `installing` state
 - {{domxref("ServiceWorkerRegistration.waiting")}} — when the service worker is in `installed` state
```

File: files/en-us/web/api/serviceworkerglobalscope/clients/index.md [hunk 6]
```diff
@@ -16,6 +16,12 @@ object associated with the service worker.
 
 The {{domxref("Clients")}} object associated with the specific worker.
 
+## Syntax
+
+```js-nolint
+self.clients
+```
```
Comment (hamishwillee) [inline files/en-us/web/api/serviceworkerglobalscope/clients/index.md:23]: No, properties just have "Value" for a syntax section - https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Page_types/API_property_subpage_template

File: files/en-us/web/api/serviceworkerglobalscope/clients/index.md [hunk 7]
```diff
@@ -16,6 +16,12 @@ object associated with the service worker.
...
+
 ## Specifications
```
Comment (hamishwillee) [inline files/en-us/web/api/serviceworkerglobalscope/clients/index.md:25]:
```suggestion
The {{domxref("Clients")}} object associated with the specific worker.

## Specifications
```

File: files/en-us/web/api/serviceworkerglobalscope/registration/index.md [hunk 8]
```diff
@@ -14,6 +14,12 @@ The **`registration`** read-only property of the {{domxref("ServiceWorkerGlobalS
 
 A {{domxref("ServiceWorkerRegistration")}} object.
 
+## Syntax
+
+```js-nolint
+self.registration
+```
+
 ## Specifications
```
Comment (hamishwillee) [inline files/en-us/web/api/serviceworkerglobalscope/registration/index.md:23]: No, just the value section.

```suggestion
A {{domxref("ServiceWorkerRegistration")}} object.

## Specifications
```

File: files/en-us/web/api/serviceworkerglobalscope/serviceworker/index.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+A {{domxref("ServiceWorker")}} object.
+
+## Syntax
+
+```js-nolint
+self.serviceWorker
+```
+
+## Specifications
```
Comment (hamishwillee) [inline files/en-us/web/api/serviceworkerglobalscope/serviceworker/index.md:23]: No syntax section for properties

```suggestion
A {{domxref("ServiceWorker")}} object.

## Specifications
```

File: files/en-us/web/api/serviceworkerglobalscope/serviceworker/index.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+{{Compat}}
+
+## See also
+
+- [Using Service Workers](/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers)
+- [Service workers basic code example](https://github.com/mdn/dom-examples/tree/main/service-worker/simple-service-worker)
+- [Is ServiceWorker ready?](https://jakearchibald.github.io/isserviceworkerready/)
+- {{jsxref("Promise")}}
+- [Using web workers](/en-US/docs/Web/API/Web_Workers_API/Using_web_workers)
```
Comment (hamishwillee) [inline files/en-us/web/api/serviceworkerglobalscope/serviceworker/index.md:31]: I think perhaps just keep the most relevant doc linked.
```suggestion
- [Using Service Workers](/en-US/docs/Web/API/Service_Worker_API/Using_Service_Workers)
```

**Review by hamishwillee — APPROVED** [review]
Thanks @skyclouds2001 . I removed the Syntax sections - not used as that is "Value" for properties. Decided you were right - we can live without an example for a trivial getter. Merging.

**Comment by hamishwillee** [comment]
PS Sorry for the delay. Finally!

**Pull request merged**